cluster_tools/schedulers/slurm.py

```diff
@@ -42,13 +42,28 @@ class SlurmExecutor(ClusterExecutor):
     def get_job_array_index():
         return os.environ.get("SLURM_ARRAY_TASK_ID", None)
 
+    @staticmethod
+    def get_job_array_id():
+        return os.environ.get("SLURM_ARRAY_JOB_ID", None)
+
     @staticmethod
     def get_current_job_id():
         return os.environ.get("SLURM_JOB_ID")
 
-    def format_log_file_name(self, jobid):
+    @staticmethod
+    def format_log_file_name(jobid):
         return "slurmpy.stdout.{}.log".format(str(jobid))
 
+    @classmethod
+    def get_log_file_path(cls, cfut_dir):
+        job_id = cls.get_current_job_id()
+        job_array_id = cls.get_job_array_id()
+        job_array_index = cls.get_job_array_index()
+        return cls.format_log_file_path(
+            cfut_dir,
+            job_id if job_array_index is None else f"{job_array_id}_{job_array_index}",
+        )
```

Comment:
hmm, I think, it's suboptimal that this function essentially needs to be kept in sync with the code in line 90. ideally, there would be an ENV variable (set by slurm) from where one could read the output file handle. I briefly searched for such an env var, but couldn't find anything :(
at the very least, there should be a comment about the relation to the code in line 90 (maybe that could even be extracted into an own variable so that it's easier to explain the relation in a code comment).

Reply:
I also couldn't find an env variable that contains the output file information :/
I agree that the relation should be documented in the code and will add that. :+1: